Update release note for type narrowing change (#992)
ballerina-platform/ballerina-release: release-notes/swan-lake-beta3-release-note.md

    string s = let string[] arr = [s] in arr[0]; // Compilation error.
}
```

> - Type narrowing is now reset on any assignment to a narrowed variable. Previously type narrowing was not reset if the static type of the new value was a subtype of the narrowed type. This was a deviation from the specification.

Reviewer: ```suggestion
- Type narrowing is now reset on any assignment to a narrowed variable. Previously, type narrowing was not reset if the static type of the new value was a subtype of the narrowed type. This was a deviation from the specification.
```

```ballerina
public function main() {
    int|string v = 0;

    if v is int {
        int x = v; // Valid.

>         // Now resets the type to `int|string` even though `1` belongs to `int`.

Reviewer: ```suggestion
        // Now, it resets the type to `int|string` even though `1` belongs to `int`.
```

Author: Will change to
```suggestion
        // Now, the type is reset to `int|string` even though `1` belongs to `int`.
```
